Wilderbuilds Trackball

bullwinkle3000 opened the pull request:
Hi, Josh.  As I mentioned on Etsy, here's the PR for my trackball changes and several other tweaks.  

Almost forgot, just set thumb_style to TRACKBALL_WILD to get my version of the trackball build.

elliotf commented:
Hello, wonderful people. @bullwinkle3000 's trackball let me to investigate this project, and I'm trying to build this branch.

I've altered the default src/run_config.json `thumb_style` to `TRACKBALL_WILD` but I get an empty hole in the STL when I look at the .scad it generates. Here is my diff:
```
$ git diff
diff --git a/src/run_config.json b/src/run_config.json
index 3f17d94..d55c18d 100644
--- a/src/run_config.json
+++ b/src/run_config.json
@@ -19,7 +19,7 @@
     "column_style": "standard",
     "full_last_rows": false,
     "keyboard_z_offset": 11,
-    "thumb_style": "DEFAULT",
+    "thumb_style": "TRACKBALL_WILD",
     "default_1U_cluster": true,
     "trackball_in_wall": false,
     "tbiw_ball_center_row": 0.2,
$ git log -1
commit 9f0407fdb700c9728fab973f4fa9e1cb39d02205 (HEAD -> wilder, bullwinkle/wilder)
Author: bullwinkle3000 <bullwinkle@librem.one>
Date:   Wed Nov 10 13:33:17 2021 -0500

    Slight clean
```
Writing this comment showed me that I was behind by two commits, so I did a pull and grabbed the latest commits, but now it will not build (via the `./dactyl.sh shell --session` docker image):
```
(base) root@60d3fc3007e8:/app/src# time python3 dactyl_manuform.py
NO CONFIGURATION SPECIFIED, USING run_config.json
Traceback (most recent call last):
  File "/app/src/dactyl_manuform.py", line 60, in <module>
    with open(os.path.join(overrides), mode='r') as fid:
FileNotFoundError: [Errno 2] No such file or directory: ''

real    0m0.134s
user    0m0.271s
sys     0m0.333s
```
and it looks like it's 38a02c8598f0088eb7385547af3ea9b4b32da447 because if I change `overrides` to `null` instead of `""` it builds again, but I get a slightly different empty hole in the .scad file:

![2021-11-14T12-58-42](https://user-images.githubusercontent.com/22679/141694623-f3006852-0719-420f-9047-4fd7b752c112.png)

Is there a sample run_config.json that will generate correctly, and/or is there something missing that wasn't committed? Thank you very much for your work, and any help you can provide!

bullwinkle3000 commented:
Ah, that's right... I've been updating that branch heavily since submitting this PR.  Forgot it was awaiting approval/merge.  I'll update and fix the conflicts this week.  I might need to pull out this PR if I can't get to it and resubmit later.

Otherwise, that's an issue with .scad files I've had for a while and I haven't addressed yet as I always do my full generation in cadquery (which now exports STLs in addition to STEP files).

I've been focused mostly on my own workflow for my store working smoothly so, yes, some bits may not work out-of-the-box for users just downloading.  But feel free to ping me for help and I'll either fix the code or help you get it working.  I never use the docker or the dactyl.sh scripts so I'm unsure if those CAN work with that branch.  I'd suggest using my fork and not Josh's.  We'll eventually sync up, but there's a slew of differences now.

As for the null/"" thing, I'll fix that this afternoon and update.  I now always specify an "overrides" parameter pointing to another json config file which simply has a subset of the run_config.json params to override for a particular customer (so as not to dirty up the run_config.json).  Here's an example:

`{
  "ENGINE": "cadquery",
  "override_name": "TEST",
  "save_dir": "/some_dir_for_exported_things",
  "iteration": "please_god_just_work",
  "show_caps": false,
  "nrows": 6,
  "ncols": 6,
  "plate_style": "NOTCH",
  "tenting_angle": 0.32,
  "keyboard_z_offset": 13,
  "full_last_rows": true,
  "thumb_style": "TRACKBALL_WILD",
  "other_thumb": "DEFAULT",
  "ball_side": "right"
}`

bullwinkle3000 commented:
Lastly, I should mention my version also spits out a "freecad.py" script that, when copy/pasted into FreeCAD's python console, will import the STEP files and export STLs into the same target directory.  I generally use these for my prints as they're more lightweight than the Cadquery exports and seem to be free from errors.

elliotf commented:
Very interesting. The feedback loop using cadquery is very very slow, which is why I was using solid/openscad (I can see changes within ~2 seconds)

When you say "your version" are you referring to https://github.com/bullwinkle3000/dactyl-keyboard/tree/master or https://github.com/bullwinkle3000/dactyl-keyboard/tree/wilder ?

What command(s) do you use to generate the output that includes the `freecad.py` script?

elliotf commented:
It looks like resetting to 9e314772161a1f0ac45161f425481dd160b8470c allows me to correctly generate .stl's from openscad (for anyone else following along)

bullwinkle3000 commented:
I set up miniconda (I think the conda.sh script should work fine?) and then, within the its dactyl-keyboard conda environment, I either run it natively within my IDE (PyCharm) or at the command line from within the src directory:  python dactyl_manuform.py

Interesting re: that commit.  I'll check that out, thanks!

bullwinkle3000 commented:
Oh, yes, "my version" is usually the wilder branch on my repo.  I put all my latest changes there, though I did recently pull them into the master there.  I need to merge in Josh's recent changes as much as I can still.

bullwinkle3000 committed:
Removed null on override name

bullwinkle3000 committed:
Merge from bulk_build.

bullwinkle3000 committed:
Anonymize

bullwinkle3000 committed:
Tweaked overrides handling.  Builds should work with default run_config.json

bullwinkle3000 committed:
Replaced wonky trackball_sensor_mount.stl with original.  Openscad seems happy now.

bullwinkle3000 commented:
Just did another merge from a feature branch, bulk_build, into wilder.  It allows for batch processing of multiple board sizes and options at once (rather specifically geared towards my print shop).

Also, the two issues that @elliotf found should be fixed: The crash on "" instead of null, and the completely absent trackball socket in openscad.

joshreve commented:
I'm going to retarget this to a refactor branch (based on dev) and start the merge.  I like the cluster separation and may even extend to trying to eliminate globals.  Even if I pass all the variables in a big pile, it at least makes them IDE visible.  I also want to remover my * imports.  Another PR has some of this work and I am going to try to combine and clean up the concepts to bring them together.  I still have to sort through all the changes as it looks like you have some other things going on as well.  I think I'm going to start moving some of the extra stuff like the FreeCAD script generator to a contrib folder.  

I like a lot of what people are doing with added features, but just can't maintain them and need them in an area that designates them as unsupported features that people can try, but may or may not work long term.  I've had some similar requests from others and am already having issues maintaining the current workflows.

joshreve commented:
Merged with refactor branch.  Doesn't seem to want to update here.  I may close with comment and we can reopen a PR against refactor to continue working.  Let me know what you think about that approach.  It didn't make sense to push this to master since there it will need a bit of work to align with changes there.  We can move it to dev once major refactoring is done and to master when it is clean and running smoothly.

bullwinkle3000 committed:
Trackball cutter wasn't working, fixed now.

bullwinkle3000 commented:
Moving bits into a contrib folder is totally fine with me.  A lot of the side bits from my branch are mostly focused on my store or print reliability and haven't always been geared towards keeping a clean repo for others to check out.

Along those lines, I'm also fine with reopening the PR on refactor and keeping the tectonic shifts there.

I've written a bulk_build.py to build about 40 or so different feature combinations as I'm setting up a queuing system for incoming orders that should just pluck the needed STLs for printing rather than run a full build.  In putting that together, I had to rework the main script into a single function, so it's now vectoring towards its own class and I hope to have that in place by the beginning of next week and may have a bunch of the globals reined in.  Might be interesting to discuss possible architectures for all the various options and features since they soon needn't sit in a single massive file (I hope.).  

I have this urge to wholly shift the code into a more modular, clear, and maintainable setup if for no other reason than to entirely transition alway from the Clojure origins.  Those origins are pretty damn cool, but awfully hard to expand upon.  So many stale forks with good intentions.

bullwinkle3000 committed:
Safeguarding state

bullwinkle3000 committed:
Working on Loligagger holder code

bullwinkle3000 committed:
Key rotations off on wilder cluster

bullwinkle3000 committed:
usb_holder finishes, but has issues.

bullwinkle3000 committed:
Added --save_path argument to customize save paths on local system without dirtying json with local paths.

The overrides variable in the config now simply specifies an [override_value] directory in the save_path that contains an [override_value].json file which, coincidentally, overrides the values in the run_config.json to limit accumulation of cruft and to better target customized builds.

joshreve commented:
FYI, I have completed the first round of refactoring and have the whole lot generating again.  I'm going to look at your new additions since I pulled for merging.  After that, I should be aligned with any new features you have.  In addition, I'm planning to pull the OLEDs and probably switch plates to packages similar to the clusters.  I managed to eliminate all globals from the code which is far more stable and predictable.

Now that I have removed all globals and have scoped class and instance variables, refactoring has become much easier since the IDE actually knows where it is getting information and the normal refactoring tools can properly inspect the code.  The refactor branch is still changing so it is probably not a good choice for use yet, but I am getting close.  

@bullwinkle3000 **I would like to catch up in a week or two to see if we can realign the branches to make cross-PRs feasible again.  Some of the changes I made are drastic enough that a simple diff won't be sufficient.**  If you prefer to keep yours fully independent or don't like the other modification, I understand, but it would be good to be able to take advantage of each others work without extensive manual operations.  At that point I would close the PR and we can reopen new ones for specific upgrades.

joshreve commented:
I just reviewed the latest and it seems like primarily file controls and the USB / controller external holder so I think we are close.  I will try to pull the holder in and make it match.  For the files, I changed a bit and I am going to look at what we both have and compare.

One thing I am targeting is that the config JSONs only need parameters they want to change and I plan to actually lock down the default config, possibly with a version in the name so if I need to change it, the old one can still be referenced.  Good news is that it looks like I caught the majority of your updates with my last pull.

I still have to look at reattaching the new code in refactor to the workflow functions, but it should be relatively similar so I don't think it will take a lot to fix.

bullwinkle3000 commented:
I just saw this so figured I'd respond while I'm nose in the code.

About three weeks ago, I got about halfway through a first pass at merging my changes with the latest on yours and then shelved it all during a rush of orders.  I've since been digging deeper into cadquery and that's taken most of my current focus.  Given all the changes you've made, I think a solid merging would need to restart prior to my batch run code or we'd sort of be merging two reinvented wheels.  Any of the changes I've made since to the dactyl_manuform class have been for efficiency and to support my workflow.  Ultimately, YES, I'd love to catch up in a week or so to try to keep things workable.  

I dove into the cadquery rabbit hole by starting in on some programmatic supports to ease 3D printing issues.  The supports, I think, might work out well to limit print times, filament usage, and, I hope, even bump print quality up a notch.  Might even be exportable as a separate model for soluble supports.  Now that I'm more familiar with cadquery's api, I'm looking to bump up the quality and appearance of the models and that has me wondering if keeping the code happy for multiple engines would be feasible?  Could farm out some of the generation to engine specific scripts, but what I'm considering could diverge things even further, though I'll try to minimize the impact and keep the code clean.  There's a point at which the everything-plus-the-kitchen-sink approach goes exponential.  

I saw a lot of the work you've been doing on PCBs, skeletons, modularity and separable thumb clusters (which I've hoped to get to at some point.)  I need to check them all out more thoroughly when I have time, but they're all intriguing!

bullwinkle3000 commented:
Ah, the holder isn't actually working yet.  I'd just very roughly converted it from an openscad source from Derek's code here:  https://github.com/dereknheiley/dactyl-manuform-tight/blob/master/things/usb_holder.scad

bullwinkle3000 commented:
I've had a couple of people reach out about running under Windows and I may put together a .bat file script to build a conda environment.

joshreve commented:
On the engines, I keep mulling that over.  My one big concern is that cadquery is not a major project and could be dropped at some point.  I like the generally agnostic commands, but it doesn't work everywhere.  I already had to go engine specific with the base plate.  My intent is to at least keep an abstraction layer to the engine, but I think if there were clear reasons to make one "supported" and the other fall behind it may be worth it.  Worth further discussion once a branch exists that shows one engine has 
a huge advantage.

For the looking nicer, I think cadquery may be the way to go if the mesh smoothing tools work well.  I'm thinking something similar to the blender dactyl.  It would require generation reordering, but I already converted a lot of the geometry to do that.  I'm also going to revisit the blender version, but it has a bit of an odd construction with a lot of function arrays that will require some time to break down.

I think it makes sense for us to finish our current workstreams and evaluate.  I agree that doing so in the middle will make a mess.  Even if you diverge further, we can look at merging, but I think it will be easier when both are essentially feature complete.

bullwinkle3000 committed:
Was missing this stl

bullwinkle3000 committed:
Left cluster wasn't getting proper screw holes.

bullwinkle3000 committed:
Trackball mr keyhole had a wall cutting through it.

bullwinkle3000 committed:
Fixed bulk_build.py

bullwinkle3000 committed:
Carbonfet wasn't working.
Tweaks for bulk processing.

bullwinkle3000 committed:
Removed potential hole and blocking wall in Wilder cluster.

bullwinkle3000 committed:
OLED adjustments

bullwinkle3000 committed:
Added .ipynb_checkpoints to .gitignore

bullwinkle3000 committed:
Missing assets

bullwinkle3000 committed:
First pass at integrated wrist rests

bullwinkle3000 committed:
BTUs sorta in place

bullwinkle3000 committed:
Adjusted screws again

bullwinkle3000 committed:
BTU edits

bullwinkle3000 committed:
BTU updates, still needs support

bullwinkle3000 committed:
New BTU cluster (based on wilder version only for now)

bullwinkle3000 committed:
New BTU cluster (based on wilder version only for now)

bullwinkle3000 committed:
Merged from wilder branch.

Updated README.md with general changes and run instructions.

bullwinkle3000 committed:
Keep testing override in config

bullwinkle3000 committed:
Lowered mini cluster by 4mm

bullwinkle3000 committed:
Lowered mini cluster by 4mm

bullwinkle3000 committed:
Updated hot swap socket

bullwinkle3000 committed:
Redundant code

bullwinkle3000 committed:
Logo import cherry picked

bullwinkle3000 committed:
Fixed trackball_in_wall not working

bullwinkle3000 committed:
Back to derek_hotswap

bullwinkle3000 committed:
added quick_render option

bullwinkle3000 committed:
Adjusted wild cluster looking pretty good

bullwinkle3000 committed:
Screw hole in space fixed

bullwinkle3000 committed:
Added logo_file: null to run

bullwinkle3000 committed:
null isn't None, eh? Oops.

bullwinkle3000 committed:
Logo and screw offsets params and always generate left half items, never simply mirror the right.

bullwinkle3000 committed:
Reverting hotswap

bullwinkle3000 committed:
Simpler trackball clsuter checks

bullwinkle3000 committed:
Added 'resin' parameter which short-circuits trackball supports if true.

bullwinkle3000 committed:
Widened controller holder

bullwinkle3000 committed:
Screw holes, insert holes, and external holder width adjusted.

bullwinkle3000 committed:
Tightened width of external holder.

bullwinkle3000 committed:
Dang overrides...

bullwinkle3000 committed:
Added some mounts, basic Elite C "lolligagger" mounts and blackpill external mounts.

bullwinkle3000 committed:
Merged from new_sensor_holder.

bullwinkle3000 committed:
Oops.  Forgot da step file.

bullwinkle3000 committed:
Added amoeba mounts

bullwinkle3000 committed:
More space on carbonfet top switch hole.
Fixed an issue with latest Cadquery

bullwinkle3000 committed:
Fixed issue with cluster adding funky wall with full_last_rows option.

bullwinkle3000 committed:
Updated readme to use Python 3.8 and cadquery=master

bullwinkle3000 committed:
Updated readme to use Python 3.8 and cadquery=master

bullwinkle3000 committed:
Another carbonfet tweak

bullwinkle3000 committed:
Merge remote-tracking branch 'origin/master' into wilder

bullwinkle3000 committed:
Added standard loligaggers

bullwinkle3000 committed:
Added standard loligaggers

bullwinkle3000 committed:
Amoeba mount working well

bullwinkle3000 committed:
Merge branch 'wilder'

bullwinkle3000 committed:
Python 3.8 and CQ master

bullwinkle3000 committed:
Adjusted paths to allow run from base directory

bullwinkle3000 committed:
Update README.md with adjusted run path.

bullwinkle3000 committed:
Test paths output

bullwinkle3000 committed:
Test paths output 2

bullwinkle3000 committed:
Test paths output 3

bullwinkle3000 committed:
Direct path changes nothing, rolling back.

bullwinkle3000 committed:
Handle crazy sizes better

bullwinkle3000 committed:
sorta

bullwinkle3000 committed:
magnets attempt 1

bullwinkle3000 committed:
Horizontal_OLED

bullwinkle3000 committed:
Horizontal_OLED

bullwinkle3000 committed:
Added convenience JSON files and references for common options.

bullwinkle3000 committed:
Cut OLED section out of run_config.json

bullwinkle3000 committed:
Builds to do

bullwinkle3000 committed:
Rotated socket to keep bearing at the bottom

bullwinkle3000 committed:
Added better offsets around top trackball wild key, shifted cluster forward 1mm and top key back 1mm.

bullwinkle3000 committed:
Removed post_adj calculations on web_post methods.  Added off_w, off_h, and off_z.

bullwinkle3000 committed:
Higher for Jack

bullwinkle3000 committed:
Empty overrides

bullwinkle3000 committed:
Merge branch 'wilder' into jsonloader

bullwinkle3000 committed:
Update screws and apply updates from wilder branch

bullwinkle3000 committed:
Merged from jsonloader

bullwinkle3000 committed:
Merged from trackball_three

bullwinkle3000 committed:
tiny extra space

bullwinkle3000 committed:
Forgot to remove name

bullwinkle3000 committed:
Stronger magnet supports

bullwinkle3000 committed:
Added branch detection for specific build needs.

bullwinkle3000 committed:
Added tweaks.

bullwinkle3000 committed:
Deeper magnet slots

bullwinkle3000 committed:
Tinkered with BTU socket and trackball three.

bullwinkle3000 committed:
Rollback BTU

bullwinkle3000 committed:
Tweaked BTU socket

bullwinkle3000 committed:
Tinkering in single_plate, MXLEDBIT, line 301

bullwinkle3000 committed:
Updated bulk build

bullwinkle3000 committed:
Overrides unset

bullwinkle3000 committed:
Added a flex_pcb.json to simply handle shorter key distances.

bullwinkle3000 committed:
Fixed the damn magnet depth, dammit and damn.

bullwinkle3000 committed:
Reverted btu trackball model, for now

bullwinkle3000 committed:
Updated BTU socket

bullwinkle3000 committed:
Updated BTU socket

bullwinkle3000 committed:
Adjusted magnet offsets for plate and case

bullwinkle3000 committed:
Merged shift_column config parameter code from development branch.  Better calculations on walls and allows thumb cluster to shift left and right by X columns.

bullwinkle3000 committed:
Missing param fix

bullwinkle3000 committed:
Slight tweak to trackball cluster placement, 1mm further forward, and put all trackball-related funcs in the same damn spot in dactyl_manuform.py.

bullwinkle3000 committed:
Beta btu trackball

bullwinkle3000 committed:
Generated better all-row models

bullwinkle3000 committed:
Updated bulk_build

bullwinkle3000 committed:
Adding USB-C mount point

bullwinkle3000 committed:
Fixed shift on Minithicc cluster

bullwinkle3000 committed:
Reverted magnet depths

bullwinkle3000 committed:
USB-C and TRRS mounts

bullwinkle3000 committed:
Merge branch 'feature_connectors' into wilder

bullwinkle3000 committed:
First pass at USB-C and TRRS mount points.

bullwinkle3000 committed:
Removed need for "flex_pcb.json" and set extra_height to 0.5 all around.

bullwinkle3000 committed:
Reverted Trackball-three to bearings trackball.

Updated btu socket to phat_btu_socket_w_access and phat_btu_socket_cutter.

bullwinkle3000 committed:
Reverted Trackball-three to bearings trackball.

Updated btu socket to phat_btu_socket_w_access and phat_btu_socket_cutter.

bullwinkle3000 committed:
Ensure 3-row versions are handled for OLED mounts properly

bullwinkle3000 committed:
Doh

bullwinkle3000 committed:
Blockerize bottoms option

bullwinkle3000 committed:
Walls generated once only

bullwinkle3000 committed:
Multiple fixes for weird walls, column shift, and full rows handling.

bullwinkle3000 committed:
column_shift adjustments on cluster angles and offsets.

bullwinkle3000 committed:
Thicker front wall on wide left wall with carbonfet

Fixed screw diameter and top depth

bullwinkle3000 committed:
Adjustments to default cluster and column shift

bullwinkle3000 committed:
Preparing to drop magnets

bullwinkle3000 committed:
whole matrix still bent

bullwinkle3000 committed:
Slight bump to brass insert radii

bullwinkle3000 committed:
Adjustments to inserts and sphere tops

bullwinkle3000 committed:
Shifted right screwholes a bit

bullwinkle3000 committed:
Updated generate_configuration with latest config settings.

Added gizmo to report git commit and branch on cadquery generation.

bullwinkle3000 committed:
gitpython dependency added

bullwinkle3000 committed:
commented bit for lower version for anand_2

bullwinkle3000 committed:
yuki

bullwinkle3000 committed:
Centered logo

bullwinkle3000 committed:
Centered logo

bullwinkle3000 committed:
Fixed lego bottoms

bullwinkle3000 committed:
Added nicenano_sliding.json

bullwinkle3000 committed:
Reverted default cluster screw position

bullwinkle3000 committed:
Export each half with its own cluster name

bullwinkle3000 committed:
Freakin' magnet bottom conditional...

bullwinkle3000 committed:
Inserts, inserts

bullwinkle3000 committed:
Fixed logo

bullwinkle3000 committed:
Added encoder in wall bits.

bullwinkle3000 committed:
Start of clusters reading data from JSON

bullwinkle3000 committed:
Wider insert bores

bullwinkle3000 committed:
Revert to standard insert size.

bullwinkle3000 committed:
OLED per side working

bullwinkle3000 committed:
Shifted trackball for polydactyl

bullwinkle3000 committed:
adding choc

bullwinkle3000 committed:
Tweaks

bullwinkle3000 committed:
Adjusted polydactyl

bullwinkle3000 committed:
Fixed external breakout on bottom plate

bullwinkle3000 committed:
Encoder in wall

bullwinkle3000 committed:
More 5x7 encoder tweaks

bullwinkle3000 committed:
Removing horizontal OLED option

bullwinkle3000 committed:
Added skip keys and tweaked row handling for encoder in wall.

bullwinkle3000 committed:
Matrix funcs

bullwinkle3000 committed:
Added 0.96 inch oled first pass.

bullwinkle3000 committed:
updates to polydactyl

bullwinkle3000 committed:
Tweaked

bullwinkle3000 committed:
tweaks to 0.96 oled placement.

bullwinkle3000 committed:
Adding encoder in wall fields

bullwinkle3000 committed:
pulls for wheel encoder from trackball_in_wall

bullwinkle3000 committed:
increased size of wheel hole

bullwinkle3000 committed:
Adjusted things a little wider for encoder and ease of access on Polydactyl

bullwinkle3000 committed:
Tweaked the wheel mount a bit more.

bullwinkle3000 committed:
Fixed outer wall calculation

bullwinkle3000 committed:
Updated more complete naming and added names and tweaks to config options

bullwinkle3000 committed:
Encoder and oled adjustments

bullwinkle3000 committed:
nice!view mount too skinny...  Added 1.25mm

bullwinkle3000 committed:
maybe FIXED the minithicc issue.

bullwinkle3000 committed:
renders and nice!view fixes

bullwinkle3000 committed:
Fixes for nice!view

bullwinkle3000 committed:
Tweaks

bullwinkle3000 committed:
Tweaked the plate rendering on key switch plates.

bullwinkle3000 committed:
tweaks for encoder

bullwinkle3000 committed:
Added specific size files

bullwinkle3000 committed:
Added TRACBALL_ORBYL5

bullwinkle3000 committed:
Adjusted screw on Minithicc3

bullwinkle3000 committed:
Remove auto render of pngs

bullwinkle3000 committed:
Had to adjust thumb screw on default cluster

bullwinkle3000 committed:
Screws whacky on thumb clusters...

bullwinkle3000 committed:
Remove trackball_in_wall setting on wilder branch size json files.

bullwinkle3000 committed:
Tweaks to encoder positions

bullwinkle3000 committed:
Tweaks to screw positions

bullwinkle3000 committed:
Adjustements to 6x7

bullwinkle3000 committed:
Pulled updates from testing branch

bullwinkle3000 committed:
smaller cutout

bullwinkle3000 committed:
adjustments to trackball_orbyl

bullwinkle3000 committed:
rotation on trackball wild not great

bullwinkle3000 committed:
Trying btu + ceramic socket

bullwinkle3000 committed:
wrist rest rendering

bullwinkle3000 committed:
PLAYING WITH WRIST REST

bullwinkle3000 committed:
tweaks to 6x7 standard

bullwinkle3000 committed:
tweaks to 6x7 and 5x6

bullwinkle3000 committed:
Rotate left side on resin and trackball adjustments

bullwinkle3000 committed:
adjusted carbonfet to add another 1.5u top key

bullwinkle3000 committed:
Updated controller holder with screw mounts

bullwinkle3000 committed:
Corrected side of screws on controller mount

bullwinkle3000 committed:
Adjustments to OLED

bullwinkle3000 committed:
Preparing to add color to config for renders

bullwinkle3000 committed:
Adjusted screw for trackball wild

bullwinkle3000 committed:
trackball_in_wall false

bullwinkle3000 committed:
carbonfet broke rendering

bullwinkle3000 committed:
Missing 6x6 files

bullwinkle3000 committed:
good sockets

bullwinkle3000 committed:
good sockets

bullwinkle3000 committed:
Tweaks to 6x7 standard

bullwinkle3000 committed:
Fix for breakout and tighter spacing

bullwinkle3000 committed:
Adding ili9341.json for touch screen

bullwinkle3000 committed:
Tweaks to il98341

bullwinkle3000 committed:
Adjustments to ili9341.json

bullwinkle3000 committed:
Added new USB-C clip in connector mount and configuration options.

bullwinkle3000 committed:
New usb clip mount

bullwinkle3000 committed:
JoystickWild cluster

bullwinkle3000 committed:
JoystickWild cluster

bullwinkle3000 committed:
Joystick work

bullwinkle3000 committed:
Adjustments to Nice!View undercut

bullwinkle3000 committed:
fixes to 6x7

bullwinkle3000 committed:
smaller row widths

bullwinkle3000 committed:
smaller row widths

bullwinkle3000 committed:
Tweaks to 4x6 standard

bullwinkle3000 committed:
ENCODER WAY OFF ON 5X6

bullwinkle3000 committed:
fixes to polydactyl and clusters

bullwinkle3000 committed:
FIXES FOR POLYDACTYL

bullwinkle3000 committed:
adjustments to tracbkall rotation.

bullwinkle3000 committed:
Adjustments to trackball three

bullwinkle3000 committed:
prepping for trackpoint

bullwinkle3000 committed:
prepping for trackpoint

bullwinkle3000 committed:
tightened cluster keys for chocs

bullwinkle3000 committed:
working on assimilator

bullwinkle3000 committed:
working on assimilator

bullwinkle3000 committed:
assimilator close

bullwinkle3000 committed:
assimilator

bullwinkle3000 committed:
Fix for 6x6 and encoder on left

bullwinkle3000 committed:
Reworked json overriding

bullwinkle3000 committed:
way to handle extra encoder on side.

bullwinkle3000 committed:
6x7 adjustments

bullwinkle3000 committed:
adjusted nicenano_sliding.json

bullwinkle3000 committed:
adjustment to assimilator mount

bullwinkle3000 committed:
cut assimilator with walls, if necessary.

bullwinkle3000 committed:
adjusted 4x6_standard

bullwinkle3000 committed:
fix to minidox

bullwinkle3000 committed:
tweaks to 5x7

bullwinkle3000 committed:
Added capability to print hand size prints with less walls and trackball sphere.

bullwinkle3000 committed:
new assimilator mount
